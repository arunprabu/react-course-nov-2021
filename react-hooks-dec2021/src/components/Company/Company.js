import React, { useState } from 'react';
// import './Company.css';

const Company = (props) => {
  
  const [favCompany, setFavCompany] = useState('');

  const updateCompany = (e) => {
    setFavCompany(e.target.value)
  }

  return(
    <div>
      <p>Welcome to My Company's Official Webpage!!!</p>
      <div data-testid='companyEl' className='redText'>{props.companyName}</div>
      {/* adding test case to find the following element by placeholder */}
      <input type='text' placeholder="company"/> 
      <p>This is amazing!!</p>
      <br/>
      <br/>
      <input type='text' 
        placeholder='Fav Company' 
        value={favCompany}
        onChange={updateCompany}/>
    </div>
  )
}
export default Company;