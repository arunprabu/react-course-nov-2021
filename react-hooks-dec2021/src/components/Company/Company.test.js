import React from 'react'; // needed
import ReactDOM from 'react-dom'; //needed
import Company from './Company';
import { render, cleanup, fireEvent } from '@testing-library/react'; // from react testing lib 
import renderer from 'react-test-renderer';

afterEach(cleanup); // we can cleanup the tests using after each


// group of related tests -- test cases
describe('company component', () => {

  it('renders the company comp properly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Company/>, div);
  });

  it('should display the prop correctly', () => {
    // take the element to put under test
    const { getByTestId } = render( <Company companyName='Twitter'></Company>);
    // fetching the element by companyEl; // refer Company.js 
    // checking if props are received correctly
    expect(getByTestId('companyEl')).toHaveTextContent('Twitter');// companyEl is the test id of the element
  });

  //finally let's focus on snapshot testing
  it('should have right company comp snapshot', () => {
    const tree = renderer.create(<Company name='Facebook'></Company>).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });

});