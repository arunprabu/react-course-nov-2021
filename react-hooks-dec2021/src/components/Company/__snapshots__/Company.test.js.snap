// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company component should have right company comp snapshot 1`] = `
<div>
  <p>
    Welcome to My Company's Official Webpage!!!
  </p>
  <div
    className="redText"
    data-testid="companyEl"
  />
  <input
    placeholder="company"
    type="text"
  />
  <p>
    This is amazing!!
  </p>
  <br />
  <br />
  <input
    onChange={[Function]}
    placeholder="Fav Company"
    type="text"
    value=""
  />
</div>
`;
